src/TreeView/types.ts

```diff
@@ -17,8 +16,10 @@ export type INodeRefs = null | React.RefObject<{
 export type EventCallback = <T, E>(
   event: React.MouseEvent<T, E> | React.KeyboardEvent<T>
   ) => void;
-  
-  export interface INode<M extends IFlatMetadata = IFlatMetadata> {
+
+export type IMetadata = undefined;
```

Comment:
I personally find aliases to a single primitive type to be very confusing. And this one is extra misleading since IMetadata can be anything. So i am requesting that you delete the IMetadata type since it no longer serves any purpose. The type constraint on INode is enough.

The only reason I would consider keeping INode is if there is a way to alias “this is an object but not an array”; however, I don’t believe that is currently capable in Typescript with generics.

So please delete IMetadata.